prog/rand.go

```diff
@@ -268,6 +268,20 @@ func (r *randGen) filename(s *state, typ *BufferType) string {
 	return fn
 }
 
+func (r *randGen) filenameMatched(s *state, typ *BufferType) string {
+	if len(typ.Values) == 0 {
+		return ""
+	}
+	pattern := typ.Values[0]
+	if files, ok := s.target.Files[pattern]; ok {
```

Comment:
This just selects a random value from the set. We already have this code for strings:
```
func (r *randGen) randString(s *state, t *BufferType) []byte {
	if len(t.Values) != 0 {
		return []byte(t.Values[r.Intn(len(t.Values))])
	}
```

I think we should install files for a given glob into t.Values. It looks like a nice design, we will have them local to the type rather than in-directed via target.Files.
And then we can use randString.

Then it makes sense to put the glob itself into BufferType.Subkind to make it clear that it's a value, it will prevent using the glob as value accidentially.